doc/WebRTC.xml

```diff
@@ -604,17 +621,22 @@ Device -> Server: { "method": "register", "params": {"authorization": "JWT..."},
 Server -> Device: { "result": { "id": "device-b1" }, "id": 1}
 
 Client -> Server: { "method": "connect", 
-              "params": {"authorization": "JWT...", "DeviceID": "device-b1"}, "id":2}
+                "params": {"authorization": "JWT...", "DeviceID": "device-b1"}, "id":2}
 Server -> Device: { "method": "connect", 
-                "params": {"session": "s1", "iceServers": [{"urls": ["stun:1.2.3.4"]}], "id":2}
-  Device -> Server: { "result": {}, "id": 2}
+                "params": {"session": "s1", "iceServers": [{"urls": ["stun:1.2.3.4"]}], 
+                "id":2}
+Device -> Server: { "result": {}, "id": 2}
 Server -> Client: { "result": {"session": "s1", 
-                               "iceServers": [{"urls": ["stun:1.2.3.4"]}]}, "id": 2}
+                "iceServers": [{"urls": ["stun:1.2.3.4"]}]}, "id": 2}
 
-  Device -> Server: { "method": "invite", "params": {"session": "s1", "offer": "SDP..."}, "id":3}
-Server -> Client: { "method": "invite", "params": {"session": "s1", "offer": "SDP..."}, "id":3}
-Client -> Server: { "result": {"answer": "SDP..."}, "id": 3}
-  Server -> Device: { "result": {"answer": "SDP..."}, "id": 3}
+Device -> Server: { "method": "invite", "params": {"session": "s1", "offer": "SDP...", 

```

Comment:
As per the example flow, it seems **dataChannelSubprotocols** offer/response is **outside** of SDP but as part of INVITE transaction.

But the specification text in Section Enabling data channels has the text as quoted 
  - "it shall signal this in the **SDP offer** of the invite method. If the client supports data channels and intends to use them during the WebRTC session, it shall signal this in the **SDP answer**."

May I suggest text should say 
  - it shall signal this as part of the invite method. If the client supports data channels and intends to use them during the WebRTC session, it shall signal this in the invite response along with SDP.

Reply:
There are two levels of negotiation:
1) First is standard WebRTC datachannel negotiation, which is part of the SDP offer/answer. This is required for WebRTC clients to enable data channels in the session.
2) Second is our ONVIF protocols negotiation, which is required to enable specific ONVIF subprotocols and they are not part of the SDP, but instead INVITE request-answer.

Reply:
Specification is not very clear about this 2-level negotiation, from my understanding the current example flow seems to cover/represent level-2 of negotiation process in INVITE,  Would it be possible to add some example for the level-1 negotiation in SDP offer/answer? Or is that kind of covered in WebRTC RFC?

Reply:
This is a standard way of enabling data channels covered by WebRTC W3C specs/RFCs and it usually involves calling createDataChannel() on PeerConnection object before SDP is exchanged. In my opinion the first paragraph is sufficient, it avoids referring to specific WebRTC APIs on how this can be achieved, which may be implementation specific.